### ISSUE TITLE
SSL Certificate Error

### ISSUE THREAD

**Issue by Tim-Otte** [nextcloud/passman#431]
### Steps to reproduce
1. Go to passman.cc in your webbrowser

### Expected behaviour
Show website with secure SSL connection

### Actual behaviour
Showing NET::ERR_CERT_AUTHORITY_INVALID error

### Configuration
**Operating system**:   Windows 10

**Browser**:      Chrome

**Extensions that might cause interference**:   Nope

**Passman version**:    2.1.4

**Operating system**: Linux

**Web server:** Apache

**Database:** MySQL

**PHP version:** 7.x
      
**cloud server:** Nextcloud
   
#### Browser log
<details>
<summary>Browser error log</summary>
Subject: demo.passman.cc

Issuer: Untrusted Bitdefender CA

Expires on: 14.01.2018

Current date: 15.01.2018

PEM encoded chain:
-----BEGIN CERTIFICATE-----
MIIDszCCApugAwIBAgIJAOoVf7BbKiqvMA0GCSqGSIb3DQEBCwUAMFQxITAfBgNV
BAMMGFVudHJ1c3RlZCBCaXRkZWZlbmRlciBDQTEMMAoGA1UECwwDSURTMRQwEgYD
VQQKDAtCaXRkZWZlbmRlcjELMAkGA1UEBhMCVVMwHhcNMTcxMDE1MjMzMTUxWhcN
MTgwMTEzMjMzMTUxWjAaMRgwFgYDVQQDEw9kZW1vLnBhc3NtYW4uY2MwggEiMA0G
CSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQC0vBAWIrxRo2Frh7yPjbvPl+EyNHUz
YccEKlxGF79yp2S/0RKlRC3ldf1TAiD+PIt2E8gxBrpOfy38PDPjfhDwJoZaE2no
O+aOKyt94XFQTV+HcbvfolLMzbOAenLkxwvu2+zARry2CsvRIDNH8eFeAJF3A+59
RWzfyjsGCDzHN4Tzt3KrKbZwYd3xmwKSABiwHh7h06OhIoCG78JDYXlQrseQ9gYK
Kd9fs3nc39KlXdwL+NYweb5Q4DYUxYR6kNQS8yTj6Hsm+puUzo2PmSPoGMbKC9H3
VlI0y1vyOcXeOdPpjbHQ5aakSjmbdy96UFj9Yb0X6wYTNktu5jiw5Pu1AgMBAAGj
gcEwgb4wgbsGA1UdEQSBszCBsIIPZGVtby5wYXNzbWFuLmNjgg5pbWcucGFzc21h
bi5jY4IVbWF0dGVybW9zdC5wYXNzbWFuLmNjggpwYXNzbWFuLmNjghNyZWxlYXNl
cy5wYXNzbWFuLmNjghBzdGF0cy5wYXNzbWFuLmNjgg90ZXN0LnBhc3NtYW4uY2OC
EndlYmhvb2sucGFzc21hbi5jY4IOd3d3LnBhc3NtYW4uY2OCDnpuYy5wYXNzbWFu
LmNjMA0GCSqGSIb3DQEBCwUAA4IBAQAFId3lCzy8YhDyWeCsRbson5p2u2pQ4otl
/CO3iU/wf+gIz9Tnk/AobxJD//8DWbyR++gfN2AGrfR76Jf+zb8uPeFYNtQaJxbT
h0Vy/qUZnjebAghH3Qm8R+B+qnAC8dY1/GJXzHG8H7gt6soxBd6Ws47Lwxoqmht4
vHenGhvXsUnM54Z0bg2s/ugG+zcBxWDmx5jgf510ChvQxkEIldVfYBlp2d482eUf
XWA1hekja1HKujkWsvQGLND87CXauc39DTWqJeV7vCzzM0BxjKBD2bZnC7ffr8zZ
+IZyDfQC+atSPIdYKAZU7BjagYGa5C/eiIaDGgaR/cwj97UVz3BA
-----END CERTIFICATE-----
</details>   

### Description
Please update your SSL certificate...

### Benefit / value
Much fun with SSL

### Risk / caveats
None

**Are you a developer willing to implement this feature?**: no (I can't, it's your domain)

**Can you sponsor the development of this feature or do you know someone who can?**: no (why?)

**Comment by brantje** [comment]
Yea we know that the site gives an SSL error.
This is happening because Let's Encrypt is having some issues.
More info here: https://letsencrypt.status.io

**Comment by Tim-Otte** [comment]
The plugin needs to access your website to get enabled. How can I activate it without an SSL error?

**Comment by brantje** [comment]
Afaik passman can work without the passman.cc domain. What error do you get?

**Comment by Tim-Otte** [comment]
URL error 60: SSL certificate problem: certificate has expired

**Comment by brantje** [comment]
Read my first comment. We are currently unable to obtain new certs due a bug in Let's encrypt.

**Comment by Tim-Otte** [comment]
I know what you've written. The problem is, that Nextcloud can't enable it without your website. Any ideas for a workaround?

**Comment by brantje** [comment]
Well, thats a nextcloud problem then, don't know why they ping our domain on enabling. Seems like a privacy leak.

**Comment by AnXh3L0** [comment]
This shouldn't be a Nextcloud problem either, because I use Passman on my Nextcloud server and it's working correctly. What nextcloud version do you have?

**Comment by Tim-Otte** [comment]
@brantje Okay. We'll wait for LetsEncrypt. I wonder that your domain does not work. I have no problems with my domain using LetsEncrypt.

**Comment by brantje** [comment]
That's because my certs expired last weekend. Their certs have a 3 month range, my last renewal was in november ;).

**Comment by Tim-Otte** [comment]
I wish you luck with renewing :)

**Comment by ldesgrange** [comment]
Hi,

Might be a good idea to keep this ticket open while the issue is ongoing (in order to prevent duplicates like #432 ).

Also LetsEncrypt [TLS-SNI will never come up again (only temporarily in some cases)](https://community.letsencrypt.org/t/tls-sni-challenges-disabled-for-most-new-issuance/50316), if you are using certbot, you can either wait to have a new version of certbot with HTTP-01 support or switch to an other method that is already supported by certbot like [webroot](http://letsencrypt.readthedocs.io/en/latest/using.html#webroot). HTH.

Anyway, good luck with the certificate renewal.

**Comment by brantje** [comment]
I have enabled Cloudflare for now until the problem is resolved. Certbot is on the latest version, so need some time to figure out what changed.

**Comment by brantje** [comment]
Domains should be accessible via cloudflare now. Keeping this issue open until we're back to normal operation.

**Comment by ldesgrange** [comment]
Cool, thanks. Did you put releases.passman.cc behind cloudflare too? (I'm still getting an SSL error)

**Comment by brantje** [comment]
I just checked, seems to work, maybe browser cache?

**Comment by JoergPalmer** [comment]
The app still cannot be enabled (as of January 16, 1:30pm CET) in my nextcloud instance.

Getting error message "cURL error 60: SSL certificate problem: certificate has expired"

**Comment by brantje** [comment]
That is a nextcloud problem, still don't get it why they ping the domain on install. Imho a huge privacy issue, as i can could track the installs.

**Comment by JoergPalmer** [comment]
ohh ... our postings were met themselves in the middle ;-) After deleting cache it works now. Thanks!

**Comment by ldesgrange** [comment]
Yeah, it works on a different machine, probably DNS cache issue on my machine.
When enabling Passman in Nextcloud the first time it downloads it from https://releases.passman.cc, hence the issue reported by @JoergPalmer, should be fine soon once the cache is expired.

**Comment by brantje** [comment]
Ahh makes sense then, always thought they hosted the zips.

**Comment by ontherunvaro** [comment]
Just pinging in to report that SSL is still broken for demo.passman.cc, other subdomains seem to be working fine though.

**Comment by animalillo** [comment]
Oh, right, i forgot to update the demo launch command to point to the new path after the update of lets encrypt. Will check it tomorrow unless @brantje fixes it first

**Comment by animalillo** [comment]
New created demo instances will use the new valid certificates from lets encrypt!!
I think this can be closed now unless we still have missed some service or something?

**Comment by rugk** [comment]
Next time maybe consider https://sslping.com/ to get notified of such problems in advantage…
